### PR TITLE
Toggle button enhancement for [issue #55] done!

### PR DESCRIPTION
**Pull request by SahibArora**
Hi there,

I have completed the enhancement, if you click the `I'm a teacher` button on the Home page, it will take you to the sign in page with `I'm a teacher` selected.
I had to change the initial state for `loginType` to get the things working correctly. 
Please have a look and let me know if there is any issue in it.
Thank you!

### CODE REVIEW

File: components/loginform.js [hunk 1]
```diff
@@ -3,14 +3,14 @@ import TwoItemSwitcher from './twoItemSwitcher'
...
-    super()
+  constructor (props) {
+    super(props)
     this.state = {
       name: '',
       user: '',
       pass: '',
       register: false,
-      loginType: 'donor'
+      loginType: props.type === true ? 'teacher' : 'donor'
```
Comment (stripedpajamas) [inline components/loginform.js]: If the prop is `type` then the value should be either teacher or donor, not a bool. I recommend you change this `loginType` to match `props.type` directly and change the prop to pass the query string directly.

File: pages/signinregister.js [hunk 2]
```diff
@@ -4,18 +4,23 @@ import Head from '../components/head'
...
+        <LoginForm type={props.teacher}/>
       </div>
     </div>
     <Footer />
   </div>
 )
 
+SignInSignUp.getInitialProps = async ({ query }) => {
+  console.log(query.type) // { type: 'teacher' }
+  return { teacher: query.type === 'teacher' }
```
Comment (stripedpajamas) [inline pages/signinregister.js]: So here you could actually just return `{ type: query.type }` and pass that type to the child component.

File: pages/signinregister.js [hunk 3]
```diff
@@ -4,18 +4,23 @@ import Head from '../components/head'
...
-        <LoginForm />
+        <LoginForm type={props.teacher}/>
       </div>
     </div>
     <Footer />
   </div>
 )
 
+SignInSignUp.getInitialProps = async ({ query }) => {
+  console.log(query.type) // { type: 'teacher' }
```
Comment (stripedpajamas) [inline pages/signinregister.js]: Can remove this log statement

**Comment by SahibArora** [comment]
I did the requested changes, please give it a look.
Thanks!

**Comment by stripedpajamas** [comment]
Looks great @SahibArora 

One last thing-- as you can see, the Travis build is failing because of some linting issues. If you run `npm run lint` in the project, you should be able to see them. Here's a copy of the errors I see in Travis:

```
  /home/travis/build/teacherfund/TeacherFund_next/components/loginform.js:108:1: Trailing spaces not allowed.
  /home/travis/build/teacherfund/TeacherFund_next/components/loginform.js:109:3: Block must not be padded by blank lines.
  /home/travis/build/teacherfund/TeacherFund_next/components/nav.js:85:24: There should be no space before '='
  /home/travis/build/teacherfund/TeacherFund_next/components/nav.js:85:24: There should be no space after '='
  /home/travis/build/teacherfund/TeacherFund_next/components/nav.js:86:24: There should be no space before '='
  /home/travis/build/teacherfund/TeacherFund_next/components/nav.js:86:24: There should be no space after '='
  /home/travis/build/teacherfund/TeacherFund_next/pages/signinregister.js:14:40: A space is required before closing bracket
  /home/travis/build/teacherfund/TeacherFund_next/pages/signinregister.js:22:11: Trailing spaces not allowed.
```

**Comment by joelwass** [comment]
You can run `npm run fix` to get our linter to auto fix probably most of the issues @SahibArora

**Comment by SahibArora** [comment]
Lint errors - `Fixed`.
Thanks for help!

**Pull request merged**